plugins/usbdmx/AsyncUsbTransceiverBase.cpp

```diff
@@ -67,62 +102,83 @@ bool AsyncUsbTransceiverBase::Init() {
 }
 
 void AsyncUsbTransceiverBase::CancelTransfer() {
-  if (!m_transfer) {
-    return;
+  m_suppress_continuation = true;
+
+  inflight_set::iterator i;
+  for (i = m_inflight.begin(); i != m_inflight.end(); ++i) {
+    int ret = m_adaptor->CancelTransfer(*i);
+    /* We ignore NOT_FOUND errors here because transfer cancellation
+     * naturally races with completion.
+     *
+     * This error simply means the transfer is not in the "in progress"
+     * state anymore.
+     *
+     * Even if we were to check for this before cancelling it could still
+     * happen that the transfer completes between the check and the
+     * cancellation taking effect.
+     */
+    if (ret != 0 && ret != LIBUSB_ERROR_NOT_FOUND) {
+      OLA_WARN << "libusb_cancel_transfer returned "
+               << m_adaptor->ErrorCodeToString(ret);
+    }
   }
 
-  bool canceled = false;
-  while (1) {
-    ola::thread::MutexLocker locker(&m_mutex);
-    if (m_transfer_state == IDLE || m_transfer_state == DISCONNECTED) {
-      break;
-    }
-    if (!canceled) {
-      m_suppress_continuation = true;
-      if (m_adaptor->CancelTransfer(m_transfer) == 0) {
-        canceled = true;
-      } else {
-        break;
-      }
-    }
+  while (!m_inflight.empty()) {
+    /* Wait for cancelled transfers to complete. Not waiting for the
+     * callbacks to happen is undefined behaviour according to libusb
+     * docs. */
+    m_cond.Wait(&m_mutex);
   }
+}
+
+struct libusb_transfer *AsyncUsbTransceiverBase::CurrentTransfer() {
+  /* This should never happen. In AsyncUsbSender::SendDMX we only ask the
+   * perform superclass to perform a transfer m_idle is not empty and in
```

Comment:
Typo?
```suggestion
   * perform superclass to perform a if transfer m_idle is not empty and in
```

Reply:
The actual typo is 'ask the perform superclass' to perfom' -> 'ask the superclass to perform'.

Reply:
:facepalm: Oops, yeah that makes more sense!

Reply:
Done